staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1/types.go

```diff
@@ -147,3 +154,45 @@ type CustomResourceValidation struct {
 	// OpenAPIV3Schema is the OpenAPI v3 schema to be validated against.
 	OpenAPIV3Schema *JSONSchemaProps `json:"openAPIV3Schema,omitempty" protobuf:"bytes,1,opt,name=openAPIV3Schema"`
 }
+
+// CustomResourceSubResources defines the status and scale subresources for CustomResources.
+type CustomResourceSubResources struct {
+	// Status denotes the status subresource for CustomResources
+	Status *CustomResourceSubResourceStatus `json:"status,omitempty" protobuf:"bytes,1,opt,name=status"`
+	// Scale denotes the scale subresource for CustomResources
+	Scale *CustomResourceSubResourceScale `json:"scale,omitempty" protobuf:"bytes,2,opt,name=scale"`
+}
+
+// CustomResourceSubResourceStatus defines how to serve the status subresource for CustomResources.
+// Status is represented by the `.status` JSON path inside of a CustomResource. When set,
+// * exposes a /status subresource for the custom resource
+// * PUT requests to the /status subresource take a custom resource object, and ignore changes to anything except the status stanza
+// * PUT/POST/PATCH requests to the custom resource ignore changes to the status stanza
+type CustomResourceSubResourceStatus struct{}
+
+// CustomResourceSubResourceScale defines how to serve the scale subresource for CustomResources.
+type CustomResourceSubResourceScale struct {
+	// SpecReplicasPath defines the JSON path inside of a CustomResource that corresponds to Scale.Spec.Replicas.
+	// Only JSON paths without the array notation are allowed.
+	// Must be a JSON Path under .spec.
+	// If there is no value under the given path in the CustomResource, the /scale subresource will return an error on GET.
+	SpecReplicasPath string `json:"specReplicasPath" protobuf:"bytes,1,name=specReplicasPath"`
+	// StatusReplicasPath defines the JSON path inside of a CustomResource that corresponds to Scale.Status.Replicas.
+	// Only JSON paths without the array notation are allowed.
+	// Must be a JSON Path under .status.
+	// If there is no value under the given path in the CustomResource, the status replica value in the /scale subresource
+	// will default to 0.
+	StatusReplicasPath string `json:"statusReplicasPath" protobuf:"bytes,2,opt,name=statusReplicasPath"`
+	// LabelSelectorPath defines the JSON path inside of a CustomResource that corresponds to Scale.Status.Selector.
+	// Only JSON paths without the array notation are allowed.
+	// Must be a JSON Path under .status.
+	// If there is no value under the given path in the CustomResource, the status label selector value in the /scale
+	// subresource will default to the empty string.
+	// +optional
+	LabelSelectorPath string `json:"labelSelectorPath,omitempty" protobuf:"bytes,3,opt,name=labelSelectorPath"`
```

Comment:
optional fields need to be pointers.

Reply:
Is this field actually useful without an indication of what type of object the selector should be used on?

Reply:
Or is it assumed to be pods?

Reply:
/cc @DirectXMan12

Reply:
The label selector semantics is implicit in the intended CRD semantics by the user. Assume a `MachineSet` that our cluster api colleagues are building to scale a cluster. The label selector for their CRD will be about machines. So I don't think we need an indication of the object here.

Reply:
> Is this field actually useful without an indication of what type of object the selector should be used on?

There should be a comment that it's for pods, but yeah, the scale interface assumes it to be pods.

> optional fields need to be pointers.

I'm still a bit torn on the optionality of this field in general -- HPA won't work if this field isn't set.  Since the docs explicitly say "this will yield an empty label selector if there's no value found", why does this need to be optional in the first place?

Reply:
> HPA won't work if this field isn't set

But this is only about HPA in cases like implementing another ReplicaSet-kind object. There can be completely unrelated scalable resources and completely unrelated HPA-like autoscalers.

Reply:
You're not wrong, and going back and checking, the labelSelector status field is optional.  Add a comment about the HPA use case and I'm good.